lib/Listener/UserDeleted.php

```diff
@@ -21,23 +21,18 @@
  */
 class UserDeleted implements IEventListener {
 
-	/** @var TotpSecretMapper */
-	private $totpSecretMapper;
-
-	/** @var LoggerInterface */
-	private $logger;
-
-	public function __construct(TotpSecretMapper $totpSecretMapper, LoggerInterface $logger) {
-		$this->totpSecretMapper = $totpSecretMapper;
-		$this->logger = $logger;
+	public function __construct(
+		private TotpSecretMapper $totpSecretMapper,
+		private LoggerInterface $logger,
+	) {
 	}
 
 	public function handle(Event $event): void {
 		if ($event instanceof UserDeletedEvent) {
 			try {
-				$this->totpSecretMapper->deleteSecretByUserId($event->getUser()->getUID());
+				$this->totpSecretMapper->deleteSecretByUserId($event->getUser()?->getUID());
```

Comment:
`\OCP\User\Events\UserDeletedEvent::getUser` can not return `null`. If it could, we'd get a type error for `\OCA\TwoFactorTOTP\Db\TotpSecretMapper::deleteSecretByUserId` being called with null.

This does not make sense